### ISSUE TITLE
TS clears MacOS DNS even without "Use TS DNS settings" (with exit node)

### ISSUE THREAD

**Issue by kim0** [tailscale/tailscale#7333]
### What is the issue?

Before starting TS, DNS is set to the local resolver `127.0.0.1`

```
$ # TS exited
$ scutil --dns | head
DNS configuration

resolver #1
  nameserver[0] : 127.0.0.1
  flags    : Request A records, Request AAAA records
  reach    : 0x00030002 (Reachable,Local Address,Directly Reachable Address)

resolver #2
  domain   : local
  options  : mdns
$ dscacheutil -q host -a name ifconfig.me
name: ifconfig.me
ip_address: 34.160.111.145
```

Starting TS with exit node, and with unchecked "Use Tailscale DNS setting" still clears my `127.0.0.1` DNS resolver and breaks all Internet access

```
$ # Start TS now. With exit node, not accepting DNS from TS. Still clears local DNS
$ scutil --dns | head
DNS configuration

resolver #1
  flags    : Request A records, Request AAAA records
  reach    : 0x00000000 (Not Reachable)

resolver #2
  domain   : local
  options  : mdns
  timeout  : 5
$ dscacheutil -q host -a name ifconfig.me
$ ping -c1 google.com
ping: cannot resolve google.com: Unknown host
```

### OS

macOS

### OS version

13.2.1

### Tailscale version

1.36.1

### Bug report

BUG-ece17d33b467b1ab46bebf3fddbf09ed7b2e0040c0cc2e22185306cb51b9e3d4-20230221001209Z-d308aeb7ceaca706

**Comment by DentonGentry** [comment]
DNS often returns geo-located results depending on where the query originates from. The intent of the exit node feature is that the node teleports to the location of the exit node, including that its DNS queries get results as though it were at the location of the exit node. This is accomplished by forwarding DNS queries to the exit node for resolution.

**Comment by DentonGentry** [comment]
As this appears to be working as intended, barring further comment we'll expect to close this during the next regular bug scrub.

**Comment by kim0** [comment]
Thanks for the reply. I don't really agree that this is expected even given the points you have mentioned. My replies are:
* `DNS returns different results depending on geo` => correct. And I want that to be from my laptop's location, not my exit node's
* `The exit node teleports you to its location` => Seen this in multiple other tickets. This really needs more thought from the team. While this is technically true, any traffic destined to the LAN, or in my case the DNS server is on localhost on my macbook, that traffic to localhost is definitely not and should not be send over the VPN. Given that, then why are we clearing the DNS server configs!
* `This means that the "Use DNS Settings" applies on the exit node` => The linux exit node has its own `--accept-dns=` option for that. So the option on my macbook cannot be controlling the exit node's behaviour. If I'm telling my laptop to not accept the DNS configuration, I do not expect it to clear my computer's config! To me this behaviour just feels wrong. I understand if TS wants to reduce the number of user visible options, but at least allow advanced users to edit some config file or so to enable such modes `TS_DONT_CLEAR_DNS` ...etc

The use-case here is that I'm a devops consultant who works with various clients and need to have a high degree of control over DNS resolution.

**Comment by dalezorz** [comment]
I have the same issue after updating to macOS 13.2.1 but I'm not using an Exit Node. I can't simply activate Tailscale without losing DNS and so need to run "sudo networksetup -setdnsservers Wi-Fi 1.1.1.1 8.8.8.8" each time I activate Tailscale in order to browse the net.

**Comment by DentonGentry** [comment]
> but I'm not using an Exit Node

Then you are running into a different problem. Please open a new issue.

**Comment by DentonGentry** [comment]
Feature request is tracked in https://github.com/tailscale/tailscale/issues/4335
